bundles/org.openhab.binding.rfxcom/src/test/java/org/openhab/binding/rfxcom/internal/messages/RFXComThermostat3MessageTest.java

```diff
@@ -43,18 +53,237 @@ public void basicBoundaryCheck() throws RFXComException {
 
     @Test
     public void testSomeMessages() throws RFXComException {
-        String hexMessage = "08420101019FAB0280";
+        testMessage("08420101019FAB0280", MERTIK__G6R_H4TB__G6R_H4T__G6R_H4T21_Z22, 1, "106411", RFXComThermostat3Message.Commands.UP, (byte) 8, OnOffType.ON, null, UpDownType.UP, StringType.valueOf("UP"));
+        testMessage("084200000000410500", MERTIK__G6R_H4T1, 0, "65", RFXComThermostat3Message.Commands.RUN_DOWN, (byte) 0, OnOffType.OFF, null, UpDownType.DOWN, StringType.valueOf("RUN_DOWN"));
+    }
+
+    private void testMessage(String hexMessage,
+                             RFXComThermostat3Message.SubType subtype,
+                             int sequenceNumber, String sensorId,
+                             RFXComThermostat3Message.Commands command,
+                             byte signalLevel,
+                             State commandChannel,
+                             State secondCommandChannel,
+                             State controlChannel,
+                             State commandStringChannel
+    ) throws RFXComException {
         byte[] message = HexUtils.hexToBytes(hexMessage);
         RFXComThermostat3Message msg = (RFXComThermostat3Message) RFXComMessageFactory.createMessage(message);
-        assertEquals("SubType", MERTIK__G6R_H4TB__G6R_H4T__G6R_H4T21_Z22, msg.subType);
-        assertEquals("Seq Number", 1, (short) (msg.seqNbr & 0xFF));
-        assertEquals("Sensor Id", "106411", msg.getDeviceId());
-        assertEquals("Command", RFXComThermostat3Message.Commands.UP, msg.command);
-        assertEquals("Signal Level", (byte) 8, msg.signalLevel);
+        assertEquals("SubType", subtype, msg.subType);
+        assertEquals("Seq Number", sequenceNumber, (short) (msg.seqNbr & 0xFF));
+        assertEquals("Sensor Id", sensorId, msg.getDeviceId());
+        assertEquals(CHANNEL_COMMAND, command, msg.command);
+        assertEquals("Signal Level", signalLevel, msg.signalLevel);
+
+        assertEquals(commandChannel, msg.convertToState(CHANNEL_COMMAND));
+        assertEquals(secondCommandChannel, msg.convertToState(CHANNEL_COMMAND_SECOND));
+        assertEquals(controlChannel, msg.convertToState(CHANNEL_CONTROL));
+        assertEquals(commandStringChannel, msg.convertToState(CHANNEL_COMMAND_STRING));
 
         byte[] decoded = msg.decodeMessage();
 
         assertEquals("Message converted back", hexMessage, HexUtils.bytesToHex(decoded));
     }
     // TODO please add tests for real messages
+
+
+    @Test
+    public void test_command_channel_on() throws RFXComUnsupportedChannelException {
```

Comment:
test methods don't follow java naming convention. I would expect:
```suggestion
    public void testCommandChannelOn() throws RFXComUnsupportedChannelException {
```

Reply:
Could be a choice to follow that there as well, have been doing it like this for years on all my other projects :-)